apps/angular/src/app/features/home/anime-table/anime-table.component.ts

```diff
@@ -1,28 +1,100 @@
+import { animeListAttribute } from '@js-camp/angular/shared/attributes/anime-list-attribute';
+import { SnackbarComponent } from '@js-camp/angular/shared/components/error-snack-bar/error-snack-bar.component';
+import { LazyLoadImageDirective } from '@js-camp/angular/shared/directives/lazy-load-image.directive';
+import { BaseFilterParams } from '@js-camp/core/models/base-filter-params';
+import { SkeletonCellComponent } from '@js-camp/angular/app/features/home/anime-table/skeleton-cell/skeleton-cell.component';
+import { NonNullableFields } from '@js-camp/core/types/non-nullable-fields';
+import { MatSortEventMapper } from '@js-camp/core/mappers/mat-sort-event.mapper';
+import { AnimeFilterParams } from '@js-camp/core/models/anime-filter-params';
 
 /** Anime Table component. */
 @Component({
 	selector: 'camp-anime-table',
 	standalone: true,
 	templateUrl: './anime-table.component.html',
 	styleUrl: './anime-table.component.css',
-	imports: [CommonModule, AsyncPipe, MatTableModule, ProgressSpinnerComponent, ErrorMessageComponent, NullablePipe],
+	imports: [
+		CommonModule,
+		MatTableModule,
+		MatSortModule,
+		ProgressSpinnerComponent,
+		NullablePipe,
+		LazyLoadImageDirective,
+		SkeletonCellComponent,
+	],
 	changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AnimeTableComponent {
-	private readonly animeService = inject(AnimeService);
+export class AnimeTableComponent implements OnInit, AfterViewInit {
+	/** Anime list. */
+	@Input({ required: true, transform: animeListAttribute })
+	public set animeList(values: readonly Anime[]) {
+		this.dataSource.data = [...values];
+	}
+
+	/** Loading status of fetching anime list. */
+	@Input({ required: true, transform: booleanAttribute })
+	public isLoading = false;
+
+	/** Error message if something went wrong fetching anime list. */
+	@Input({ required: true, transform: emptyStringAttribute })
+	public set error(value: string) {
+		if (value !== '') {
+			this.snackBar.openFromComponent(SnackbarComponent, {
+				verticalPosition: 'top',
+				horizontalPosition: 'right',
+				data: { errorMessage: value },
+			});
+		}
+	}
+
+	/**
+	 * Page paginator to store page index and page number.
+	 * Property - pageNumber: Will be converted to the zero-based page index of the displayed list of items.
+	 * Property - pageSize: Number of items to display on a page.
+	 */
+	@Input({ required: true, transform: paginatorAttribute })
+	protected readonly pagePaginator: NonNullableFields<BaseFilterParams.Paginator> | null = null;
+
+	/** Sort change event emitter. */
+	@Output()
+	public readonly sortChange = new EventEmitter<AnimeFilterParams.Sort>();
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	private readonly snackBar = inject(MatSnackBar);
+
+	/** Convert the list to MatTableDataSource to use MatSort. */
+	protected readonly dataSource = new MatTableDataSource<Anime>();
+
+	private readonly filterParamsProvider$ = inject(FILTER_PARAMS_TOKEN);
```

Comment:
Declaring private entity before protected for consistency
```suggestion
	private readonly filterParamsProvider$ = inject(FILTER_PARAMS_TOKEN);
	
	/** Convert the list to MatTableDataSource to use MatSort. */
	protected readonly dataSource = new MatTableDataSource<Anime>();
```